src/test/test-integration.ts

```diff
@@ -0,0 +1,95 @@
+import * as viewModel from "../common/view-model";
+import * as model from "../common/model";
+import * as ko from "knockout";
+import { BoardVM } from "../koclient/ko-view-model";
+import {BoardService} from "../server/BoardService";
+
+describe("Integration test between client and service", function(){
+// TODO: maybe separate this two classes in a module
+
+  class ServerMock{
+  private _boardService: BoardService;
+  private _clients: Array<App>;
+  constructor(boardService:BoardService){
+    this._boardService = boardService;
+    this._clients = new Array<App>();
+    var server = this;
+    this._boardService.sendToClient = (changes) => {
+      for(var client in server._clients){
+        server._clients[client].onMessage({patch: changes});
+      }
+    };
+  }
+
+  public addClient = (client: App) => {
+    var server = this;
+    client.sendToServer = (changes) => {
+      server._boardService.onClientMessage(changes);
+    };
+    client.onMessage(this._boardService.onClientConnection());
+    this._clients.push(client);
+  }
+}
+
+  it("When a client are making a change, this changes shouldn't be applied on the same client", function(){
+      // Arrange
+        // Prepare variables
+      var boardVM: BoardVM = new BoardVM();
+      var client = new App(boardVM);
+      var noteVM = boardVM.createNote();
+      var beginingText = "This is the begining part of the test";
+      var endingText = " and this is the ending part of the test";
+      var boardService = new BoardService();
+      var serverMock = new ServerMock(boardService);
+      serverMock.addClient(client);
+
+      // Act
+      noteVM.content(beginingText);
+      client.onInterval();
+      noteVM.content(noteVM.content() + endingText);
+      boardService.onTic();
+
+      //Assert
+      expect(noteVM.content()).toEqual(beginingText + endingText);
+    });
+
+    it("When two clients send their changes at the same time, the last change that arrive is the change that are applied", function(){
+      // Arrange
+     var note: model.Note = {
+       title: "Title",
+       content: "",
+       posX: 10,
+       posY: 10
+     };
+     var boardService = new BoardService({name: "board name", notes: {"note1": note}});
+     var serverMock = new ServerMock(boardService);
+     var boardVM1 = new BoardVM();
+     var boardVM2 = new BoardVM();
+     var client1 = new App(boardVM1);
+     var client2 = new App(boardVM2);
+     serverMock.addClient(client1);
+     serverMock.addClient(client2);
+
+    //  Act
+     note.content = "content of first client";
+     boardVM1.update({name: "board name", notes: {"note1": note}});
+     client1.onInterval();
+     note.content = "cotent of second client";
+     boardVM2.update({name: "board name", notes: {"note1": note}});
+     client2.onInterval();
+     boardService.onTic();
+
+    //  Assert
+     expect(boardVM1.toPlain().notes).toBeDefined();
+     expect(boardVM1.toPlain().notes["note1"]).toEqual(note);
```

Comment:
This assert is testing by value, right? I mean, the note object in both clients and server should be in fact different objects and should not be compared by reference, right?